src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    if (!Animal.alive) {
+      Animal.alive = [];
+    }
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(herbivore) {
+    if (herbivore instanceof Herbivore && !herbivore.hidden) {
+      herbivore.health -= 50;
+    }
+
+    if (herbivore.health === 0) {
+      const index = Animal.alive.indexOf(herbivore);
+
+      if (index > -1) {
+        Animal.alive.splice(index, 1);
+      }
+    }
```

Comment:
[CODE PERFORMANCE] - The current approach for removing dead animals from the `Animal.alive` array uses `indexOf` and `splice` inside a conditional block. This can be inefficient and should be replaced with the `filter` method, which does not require checking the index and is more declarative. Consider using `Animal.alive = Animal.alive.filter(animal => animal.health > 0);` after the health deduction to remove any dead animals.